tests/from_csv_test.py

```diff
@@ -0,0 +1,81 @@
+import glob
+import os
+
+import pytest
+import vaex
+
+path = os.path.dirname(__file__)
+csv_path = os.path.join(path, 'data', 'small3.csv')
+
+
+def test_from_csv():
+    _cleanup_generated_files()
+
+    # can read with default options
+    df = vaex.from_csv(csv_path)
+    _assert_csv_content(df, with_index=True)
+
+    # can read an empty CSV
+    df = vaex.from_csv(os.path.join(path, 'data', 'empty.csv'), copy_index=False)
+    assert len(df) == 0
+
+    # can read as chunks iterator
+    df_iterator = vaex.from_csv(csv_path, chunk_size=1, copy_index=False)
+    df1 = next(df_iterator)
+    assert len(df1) == 1
+    df2, df3 = next(df_iterator), next(df_iterator)
+    with pytest.raises(StopIteration):
+        next(df_iterator)
+    _assert_csv_content(vaex.dataframe.DataFrameConcatenated([df1, df2, df3]))
+
+
+def test_from_csv_converting_in_chunks():
+    _cleanup_generated_files()
+
+    # can read several chunks with converting, intermediate files are deleted
+    df = vaex.from_csv(csv_path, copy_index=False, chunk_size=1, convert=True)
+    _assert_csv_content(df)
+    for filename in ['small3.csv_chunk0.hdf5', 'small3.csv_chunk1.hdf5', 'small3.csv_chunk2.hdf5']:
+        assert not os.path.exists(os.path.join(path, 'data', filename))
+    assert os.path.exists(os.path.join(path, 'data', 'small3.csv.hdf5'))
+    _cleanup_generated_files()
+
+    # fails to convert if filename cannot be derived
+    with pytest.raises(ValueError, match='Cannot derive filename to use for converted HDF5 file, '
+                                         'please specify it using convert="my.csv.hdf5"'):
+        with open(csv_path) as f:
+            vaex.from_csv(f, convert=True)
+    with open(csv_path) as f:
+        converted_path = os.path.join(path, 'data', 'small3.my.csv.hdf5')
+        df = vaex.from_csv(f, convert=converted_path, copy_index=False)
+    _assert_csv_content(df)
+    assert os.path.exists(converted_path)
+    _cleanup_generated_files()
+
+    # reuses converted HDF5 file
+    vaex.from_csv(csv_path, copy_index=False, convert=True)
+    assert os.path.exists(os.path.join(path, 'data', 'small3.csv.hdf5'))
+    try:
+        os.rename(csv_path, csv_path + '_')
+        df = vaex.from_csv(csv_path, copy_index=False, convert=True)
+        _assert_csv_content(df)
+        _cleanup_generated_files()
+    except FileNotFoundError as e:
+        assert False, "vaex.from_csv tried to read from CSV file while a converted HDF5 file existed: %s" % e
+    finally:
+        os.rename(csv_path + '_', csv_path)
+
+
+def _assert_csv_content(csv_df, with_index=False):
+    assert len(csv_df) == 3
+    if with_index:
+        assert csv_df.get_column_names() == ['x', 'y', 'index']
+    else:
+        assert csv_df.get_column_names() == ['x', 'y']
+    assert (csv_df.x.values == [5, 7, 9]).all()
```

Comment:
I generally prefer:
```
df.x.tolist() == [5, 7, 9]
```
x might in the future be an arrow array, and I am not sure arrow will like comparisons to list.

Reply:
Nice, I like it more too, will apply.